thoth/storages/graph/postgres.py

```diff
@@ -3331,6 +3331,42 @@ def get_python_package_version_metadata(
 
             return formatted_result
 
+    def get_unsolved_python_packages_all_per_adviser_run(self) -> Dict[str, List[str]]:
+        """Retrieve all unsolved packages for a certain Adviser Run that need to be re run.
+
+        Examples:
+        >>> from thoth.storages import GraphDatabase
+        >>> graph = GraphDatabase()
+        >>> graph.get_unsolved_python_packages_all_per_adviser_run()
+        {'adviser-04ab56d6': ['black'], 'adviser-054ab56d6': ['black', 'numpy']}
+        """
+        with self._session_scope() as session:
+            query = (
+                session.query(AdviserRun)
+                .filter(
+                    AdviserRun.need_re_run.is_(True),
+                )
+                .join(HasUnresolved)
+                .join(PythonPackageVersionEntity)
+            ).with_entities(
+                AdviserRun.adviser_document_id,
+                PythonPackageVersionEntity.package_name
+            )
+
+            query_result = query.all()
+
+            result = {}
+            for couple in query_result:
+                if couple[0] not in result.keys():
+                    result[couple[0]] = []
+
+                result[couple[0]].append(couple[1])
+
+            if result is None:
+                logger.info(f"No Adviser Run found that need to be re run!")
```

Comment:
This message could be more suitable for consumer, I would rather keep this log message on the component level.